python/nano/src/bigdl/nano/deps/openvino/core/openvino_model.py

```diff
@@ -14,31 +14,28 @@
 # limitations under the License.
 #
 from pathlib import Path
-from openvino.inference_engine import IECore
+from openvino.runtime import Core
+from openvino.runtime.passes import Manager
 
 
 class OpenVINOModel:
     def __init__(self, ie_network: str):
         self.ie_network = None
         self.read_network(ie_network)
-        self.exec_model = None
 
     def forward_step(self, *inputs):
-        self.compile_executable(inputs)
-        inputs = dict(zip(self.exec_model.input_info, inputs))
-        return self.exec_model.infer(inputs)
-
-    def compile_executable(self, inputs):
-        input_batch_size = inputs[0].shape[0]
-        if self.exec_model and input_batch_size == self.exec_model.input_info[0].shape[0]:
-            return
-        else:
-            assert self.ie_network, "self.ie_network shouldn't be None."
-            self.ie_network.batch_size = input_batch_size
-            self.exec_model = IECore().load_network(network=self.ie_network, device_name='CPU')
+        try:
+            return self.infer_request.infer(list(inputs))
+        except RuntimeError:
+            for in_tensor, input in zip(self.infer_request.inputs, inputs):
+                in_tensor.shape == input.shape
+            return self.infer_request.infer(list(inputs))
```

Comment:
How about leaving some comments explaining the logic here?

It seems like you are catching the `RuntimeError` to recover from the shape mismatch problem. Are there other types of RuntimeError? Maybe you can also check the error message to get an exact match.

Reply:
> It seems like you are catching the `RuntimeError` to recover from the shape mismatch problem. Are there other types of RuntimeError? Maybe you can also check the error message to get an exact match.
>

Seems that with `dynamic_axes=True`, we don't have to worry about the shape now. I will remove this try-except.
>
> Also is it safe to change the shape like this? What if the user just passes the wrong input?

No need to change now. If the input is wrong, openvino will raise `RuntimeError: Can't SetBlob with name: x, because model input (shape={?,3,256,256}) and blob (shape=(3.3.512.512)) are incompatible`. I think it's quite explicit for users. Do you think we need to catch and raise another error?

Reply:
> > It seems like you are catching the `RuntimeError` to recover from the shape mismatch problem. Are there other types of RuntimeError? Maybe you can also check the error message to get an exact match.
> 
> Seems that with `dynamic_axes=True`, we don't have to worry about the shape now. I will remove this try-except.
> 
> > Also is it safe to change the shape like this? What if the user just passes the wrong input?
> 
> No need to change now. If the input is wrong, openvino will raise `RuntimeError: Can't SetBlob with name: x, because model input (shape={?,3,256,256}) and blob (shape=(3.3.512.512)) are incompatible`. I think it's quite explicit for users. Do you think we need to catch and raise another error?

I think maybe we can just remove this try-except.